jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -479,13 +493,24 @@ public JaegerSpan startManual() {
     private ScopeManager scopeManager = new ThreadLocalScopeManager();
     private BaggageRestrictionManager baggageRestrictionManager = new DefaultBaggageRestrictionManager();
     private boolean expandExceptionLogs;
+    private JaegerObjectFactory objectFactory;
 
     public Builder(String serviceName) {
       this.serviceName = checkValidServiceName(serviceName);
-      TextMapCodec textMapCodec = new TextMapCodec(false);
+      this.objectFactory = createObjectFactory();
```

Comment:
hmm, my old (mostly dead) C++ skills turn in their grave on this one. Calling virtual method from a constructor? Even if not a problem in Java it feels very non-intuitive. Maybe this?
```
public Builder(String serviceName)  {
    this(serviceName, new JaegerObjectFactory());
}

protected Builder(String serviceName, JaegerObjectFactory objectFactory)  {
   // actual ctor
}
```
Then you only need to override `createTracer` (and don't need to pass it the factory)

Reply:
Haha. Indeed you are correct! Good catch. I definitely like the idea of less additional virtual methods.

Reply:
Luckily, C++ will be much easier to change down the road if I need to. I seriously miss typedefs, templates, and even the dark magic of implicit cast operators now that I have seen how tedious it is to migrate Java code.